mobly/controllers/android_device.py

```diff
@@ -207,8 +207,15 @@ def get_instances_with_configs(configs):
             raise Error(
                 "Required value 'serial' is missing in AndroidDevice config %s."
                 % c)
-        ad = AndroidDevice(serial)
-        ad.load_config(c)
+        is_required = c.pop("required", True)
```

Comment:
It seems weird to me to mutate the config if we don't have to.
In the README we are teaching the user that attributes they add to ad config show up in the ad object. Except for this particular one? Why the inconsistency?